guides/micronaut-object-storage-oracle-cloud/micronaut-object-storage-oracle-cloud.adoc

```diff
@@ -0,0 +1,126 @@
+external:micronaut-object-storage-base/start.adoc[]
+
+common:oracle-cloud-account.adoc[]
+common:oracle-cloud-cli.adoc[]
+
+common:jq[]
+
+common:completesolution.adoc[]
+
+common:create-app-features.adoc[]
+
+== Create a bucket
+
+common:oracle-cloud-compartment-id.adoc[]
+
+=== Use the CLI to create the bucket
+
+[source,bash]
+----
+oci os bucket create --compartment-id $C --name micronaut-guide-object-storage
+----
+
+We can also use the CLI to get the Object Storage namespace:
+
+[source,bash]
+----
+export OCI_NS=$(oci os ns get | jq -r '.data')
+----
+
+Then, configure the bucket name and namespace in `application.yml`:
+
+resource:application.yml[tag=object-storage]
+
+== Controller API
+
+external:micronaut-object-storage-base/controller-api.adoc[]
+
+source:ProfilePicturesController[tags=begin-class|end-class]
+
+callout:controller[number=1,arg0=/pictures]
+callout:executes-on[2]
+<3> https://micronaut-projects.github.io/micronaut-object-storage/latest/api/io/micronaut/objectstorage/oraclecloud/OracleCloudStorageOperations.html[OracleCloudStorageOperations]
+    is the Oracle Cloud-specific interface for using object storage.
+<4> @api@/io/micronaut/http/server/util/HttpHostResolver.html[HttpHostResolver] allows you to resolve the host for a HTTP
+    request. We will use the host to build an absolute URL for the `Location` header when uploading a picture. Learn more
+    about https://docs.micronaut.io/latest/guide/#hostResolution[Host Resolution].
+
+=== Upload endpoint
+
+Implement the upload endpoint by receiving the file from the HTTP client via `CompletedFileUpload`, and the `userId` path
+parameter. Upload it to Oracle Cloud using
+https://micronaut-projects.github.io/micronaut-object-storage/latest/api/io/micronaut/objectstorage/oraclecloud/OracleCloudStorageOperations.html[OracleCloudStorageOperations],
+and then return its `ETag` in an HTTP response header to the client:
+
+source:ProfilePicturesController[tags=upload,indent=0]
+
+<1> The key represents the path under which the file will be stored.
```

Comment:
it would be good to extract most of this to callouts so that we don't have duplication between the four guides. 

callout:object-storage-key[1]